HIP-6.md

```diff
@@ -0,0 +1,138 @@
+| hip | title                   | status | author | created    |
+| --- | ----------------------- | ------ | ------ | ---------- |
+| 6   | Validator announcements | Draft  | asaj   | 2022-12-21 |
+
+### **Brief Summary / Abstract**
+
+Defines an optional specification for Hyperlane validators to announce themselves and the location of their signed checkpoints.
+
+### **Motivation**
+
+Hyperlane relayers need to aggregate validator signed checkpoints in order to deliver interchain messages.
+
+At present, relayers are manually configured with a list of validators and the location of their signed checkpoints.
+
+Relying on manual configuration creates friction to the introduction of additional validators via sovereign consensus.
+
+This HIP defines a standard for validators to announce themselves to relayers via a smart contract on the underlying blockchain.
+
+Relayers should query this contract to discover the location of validator signed checkpoints.
+
+### **Tech Spec**
+
+#### Announcment attestations
+
+The validator announcement protocol relies on validators signing attestations with the following semantic meaning:
+
+> "Checkpoints for the mailbox contract with address _m_ on chain _d_, signed by the validator with address _v_, can be found at location _l_
+
+The Hyperlane validator binary should sign the following data upon startup and write it to the storage modality in `announcement.json`.
+
+```solidity
+/**
+ * @notice Returns the digest that Hyperlane validators should sign to support
+ * the announcement protocol
+ * @param _domain The origin domain of the Mailbox being validated
+ * @param _mailbox The address of the Mailbox being validated, as bytes32
+ * @param _storageLocation Information encoding the location of signed
+ * checkpoints, specific to the storage modality (e.g. S3 bucket URL)
+ * @return The digest to EIP-191 sign
+ */
+function getDigestToSign(
+    uint32 _domain,
+    bytes32 _mailbox,
+    string calldata _storageLocation
+)
+    external
+    pure
+    returns (bytes32)
+{
+    bytes32 _domainHash = keccak256(
+        abi.encodePacked(_domain, _mailbox, "HYPERLANE")
+    );
+    return keccak256(abi.encodePacked(_domainHash, _storageLocation));
```

Comment:
Like I mentioned, I would personally advocate for supporting addition and removals in the contract right now to avoid the need to change it later, so here would probably add "add" or "remove" to the digest, but not going to block on this